airflow/providers/amazon/aws/utils/connection_wrapper.py

```diff
@@ -127,13 +139,33 @@ def __post_init__(self, conn: "Connection"):
         init_credentials = self._get_credentials(**extra)
         self.aws_access_key_id, self.aws_secret_access_key, self.aws_session_token = init_credentials
 
-        if not self.region_name and "region_name" in extra:
-            self.region_name = extra["region_name"]
-            self.log.info("Retrieving region_name=%s from %s extra.", self.region_name, self.conn_repr)
+        if not self.region_name:
+            if "region_name" in extra:
+                self.region_name = extra["region_name"]
+                self.log.info("Retrieving region_name=%s from %s extra.", self.region_name, self.conn_repr)
+            elif "region_name" in session_kwargs:
+                self.region_name = session_kwargs["region_name"]
+                self.log.info(
+                    "Retrieving region_name=%s from %s extra['session_kwargs'].",
+                    self.region_name,
+                    self.conn_repr,
+                )
+
+        if not self.verify and "verify" in extra:
+            self.verify = extra["verify"]
+            self.log.debug("Retrieving verify=%s from %s extra.", self.verify, self.conn_repr)
 
-        if "session_kwargs" in extra:
-            self.session_kwargs = extra["session_kwargs"]
-            self.log.info("Retrieving session_kwargs=%s from %s extra.", self.session_kwargs, self.conn_repr)
+        self.profile_name = extra.get("session_kwargs", {}).get("profile_name")
```

Comment:
You've already gotten `session_kwargs` on line 115, you don't need to fetch it again from `extra` right?

Also you fetch the value form session_kwargs on line 162 anyway, so I don't think you need this line at all.

Reply:
Seems like I forgot to push last changes to this PR. There is not sense to get profile by this way